server/src/main/java/org/opensearch/bootstrap/JNAAdvapi32Library.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.bootstrap;
+
+import com.sun.jna.Native;
+import com.sun.jna.Pointer;
+import com.sun.jna.ptr.IntByReference;
+import com.sun.jna.ptr.PointerByReference;
+import com.sun.jna.Structure;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.lucene.util.Constants;
+
+import java.util.List;
+
+/**
+ * Library for Windows/Advapi32
+ *
+ * @opensearch.internal
+ */
+final class JNAAdvapi32Library {
```

Comment:
Why aren't we using the JNA advapi32 wrapper?

https://github.com/java-native-access/jna/blob/030411b909d5dfd249b1df09a7f24c44babcae64/contrib/platform/src/com/sun/jna/platform/win32/Advapi32.java#L701

Reply:
> Why aren't we using the JNA advapi32 wrapper?

Because we are only using the `jna` dependency, not `jna-platform`.   See also the [`JNAKernel32Library`](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/bootstrap/JNAKernel32Library.java) and [`JNACLibrary`](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/bootstrap/JNACLibrary.java) classes which could be eliminated.

`jna-platform` is a 1.3MB JAR and also introduces a modular transitive dependency on `java.desktop` under JPMS.  Happy to switch the code to use this added dependency, but given the very limited use of library mappings, I don't think it's needed (yet).

Reply:
I don't love copy-pasting code from 3P libraries to save 1.3MB because that's how we miss fixes in the library, but I don't have a strong opinion about this. I'll hit approve on this PR and we can fix this later. How do you feel about this issue @nknize?

Reply:
> I don't love copy-pasting code from 3P libraries to save 1.3MB because that's how we miss fixes in the library

That's more likely with util-based functions, but these mappings are rather straightforward and I actually researched the original API reference to confirm correctness for these.  I do admit to copying the javadocs (derived from Win32API) from from some guy named @dblock and I don't think he made a mistake. Your opinion of him may vary though. 😁 

More seriously, though, there's a point where there are too many native mappings and we should switch to a dependency.  I don't think we're there yet (three classes with a handful of functions in them) but we are probably at least halfway there.